api/client/run.go

```diff
@@ -228,6 +224,17 @@ func (cli *DockerCli) CmdRun(args ...string) error {
 
 	//start the container
 	if err := cli.client.ContainerStart(createResponse.ID); err != nil {
+		// If we have holdHijackedConnection, we should notify
+		// holdHijackedConnection we are going to exit and wait
+		// to avoid the terminal are not restored.
+		if config.AttachStdin || config.AttachStdout || config.AttachStderr {
+			select {
+			case <-exitCh:
+			default:
+				close(exitCh)
+			}
+			<-errCh
```

Comment:
@LK4D4  we already have a `Debugf` in `holdHijackedConnection` if `err` is not nil, so I think there is no need here.